src/Microsoft.AspNetCore.Server.HttpSys/AuthenticationHandler.cs

```diff
@@ -17,135 +16,77 @@ internal class AuthenticationHandler : IAuthenticationHandler
+                        _requestContext.Response.StatusCode = 403;
                     }
                     else
                     {
-                        context.NotAuthenticated();
+                        _requestContext.Response.StatusCode = 401;
+                        _customChallenges |= scheme;
                     }
-                }
+                    break;
+                default:
+                    throw new NotSupportedException(context.Behavior.ToString());
             }
-            return TaskCache.CompletedTask;
-        }
 
-        public Task ChallengeAsync(ChallengeContext context)
-        {
-            var automaticChallenge = string.Equals("Automatic", context.AuthenticationScheme, StringComparison.Ordinal);
-            foreach (var scheme in ListEnabledAuthSchemes())
-            {
-                var authScheme = scheme.ToString();
-                // Not including any auth types means it's a blanket challenge for any auth type.
-                if (automaticChallenge || string.Equals(context.AuthenticationScheme, authScheme, StringComparison.Ordinal))
-                {
-                    switch (context.Behavior)
-                    {
-                        case ChallengeBehavior.Forbidden:
-                            _requestContext.Response.StatusCode = 403;
-                            context.Accept();
-                            break;
-                        case ChallengeBehavior.Unauthorized:
-                            _requestContext.Response.StatusCode = 401;
-                            _customChallenges |= scheme;
-                            context.Accept();
-                            break;
-                        case ChallengeBehavior.Automatic:
-                            var identity = (ClaimsIdentity)_requestContext.User?.Identity;
-                            if (identity != null && identity.IsAuthenticated
-                                && (automaticChallenge || string.Equals(identity.AuthenticationType, context.AuthenticationScheme, StringComparison.Ordinal)))
-                            {
-                                _requestContext.Response.StatusCode = 403;
-                                context.Accept();
-                            }
-                            else
-                            {
-                                _requestContext.Response.StatusCode = 401;
-                                _customChallenges |= scheme;
-                                context.Accept();
-                            }
-                            break;
-                        default:
-                            throw new NotSupportedException(context.Behavior.ToString());
-                    }
-                }
-            }
             // A challenge was issued, it overrides any pre-set auth types.
             _requestContext.Response.AuthenticationChallenges = _customChallenges;
```

Comment:
This set for the _customChallenges no longers works because we have 3 different handlers all trying to challenge at the same time, before they were all the same handler instance, so _customChallenges would have all 3.  While now they are different handlers, so we end up only setting the last auth scheme.